src/domain/state.ts

```diff
@@ -322,33 +322,37 @@ export class ApplicationStateManager {
 
   updateApp(updatePart: Partial<UpdateApplication>, isReviewer: boolean) {
     const current = this.currentApplication;
-    switch (this.currentApplication.state) {
-      case 'APPROVED':
-        updateAppStateForApprovedApplication(current, updatePart, isReviewer);
-        break;
-
-      case 'REVISIONS REQUESTED':
-        updateAppStateForReturnedApplication(current, updatePart);
-        break;
-
-      case 'REVIEW':
-        // we are updating an application in review state (i.e. admin wants to a. approve, b. reject, c. request revisions)
-        if (!isReviewer) {
-          throw new Error('not allowed');
-        }
-        updateAppStateForReviewApplication(current, updatePart);
-        break;
-
-      case 'SIGN AND SUBMIT':
-        updateAppStateForSignAndSubmit(current, updatePart);
-        break;
-
-      case 'DRAFT':
-        updateAppStateForDraftApplication(current, updatePart);
-        break;
-
-      default:
-        throw new Error();
+    if (updatePart.state === 'CLOSED' && current.state !== 'REVIEW') {
```

Comment:
yeah good point. with things as they are it might be fine this way, but could be a problem if/when more changes get implemented based on app state. i'll try it using the transition pattern!